citro3d/src/render.rs

```diff
@@ -14,65 +16,68 @@ mod transfer;
 
 /// A render target for `citro3d`. Frame data will be written to this target
 /// to be rendered on the GPU and displayed on the screen.
-pub struct Target {
+pub struct Target<'screen> {
     raw: *mut citro3d_sys::C3D_RenderTarget,
-    color_format: ColorFormat,
+    // This is unused after construction, but ensures unique access to the
+    // screen this target writes to during rendering
+    _screen: RefMut<'screen, dyn Screen>,
 }
 
-impl Drop for Target {
+impl Drop for Target<'_> {
     fn drop(&mut self) {
         unsafe {
             C3D_RenderTargetDelete(self.raw);
         }
     }
 }
 
-impl Target {
+impl<'screen> Target<'screen> {
     /// Create a new render target with the specified size, color format,
     /// and depth format.
     ///
     /// # Errors
     ///
-    /// Fails if the specified sizes are invalid, or the target could not be
-    /// created.
+    /// Fails if the target could not be created.
     pub fn new(
-        width: u32,
-        height: u32,
-        color_format: ColorFormat,
-        depth_format: DepthFormat,
+        width: u16,
+        height: u16,
+        // TODO: is there a use case for a render target that isn't associated with a Screen?
```

Comment:
Reminder to finish TODO

Reply:
I guess this is more of a question than a TODO. I can think of some uses for a render target without a corresponding screen, I guess, but I'm not sure if these are supported by the `citro3d` API itself:

- rendering to a texture or something like https://bevyengine.org/news/bevy-0-8/#render-to-textures
- "headless" mode rendering (e.g. to a file, or remote server?) not sure how common this could be

In any case, I think rendering to screen will be the default by far, but I don't want to lock out the possibility of the above use cases. So I was thinking the API could be something like this?
```rs
impl<'screen> Target<'screen> {
    pub fn from_screen(
        screen: RefMut<'screen, dyn Screen>,
        depth_format: Option<DepthFormat>,
    ) -> Result<Self> {}
    
    pub fn new(
        width: u16,
        height: u16,
        color_format: ColorFormat,
        depth_format: Option<DepthFormat>,
    ) -> Result<Self> {}
}
```

Reply:
Seems like rendering to a texture is a supported possiblity: https://github.com/devkitPro/citro3d/blob/e8825650c6013440aaf077a77ce6605ba014732a/include/c3d/renderqueue.h#L63

Reply:
Cool, I'll probably do this in a separate PR just because it would be a bit larger of a feature / API change

Reply:
#15